src/App.js

```diff
@@ -18,7 +19,8 @@ const router = (
   <Provider store={store}>
     <Router history={history}>
       <Route path="/" component={ApplicationLayout}>
-        <IndexRoute component={Welcome} />
+        <IndexRoute component={Home} />
+        <Route path="*" component={NotFound} />
```

Comment:
So a bit of a nit-pick here, but if your react app responds to that, I think we'd need some server logic to properly respond with a 404 not found in this case.  Setthing that up is way beyond the scope of what I understand about server/client rendering of JSX.